Source/StdoutLogger.cpp

```diff
@@ -1,56 +1,83 @@
 #include <AndroidExtensions/StdoutLogger.h>
-#include <unistd.h>
 #include <android/log.h>
 #include <pthread.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
 
 namespace
 {
-    static int pfd[2];
-    static int fd_saved[2];
-}
+    int fd_stdout[2];
+    int fd_stderr[2];
+    int fd_saved[2];
 
-namespace android::StdoutLogger
-{
-    static void* ThreadFunc(void*)
+    void thread_func(int fd, int prio)
     {
-        ssize_t rdsz;
-        char buf[128];
-        while ((rdsz = read(pfd[0], buf, sizeof buf - 1)) > 0)
+        FILE* stream = fdopen(fd, "r");
+
+        while (true)
         {
-            if (buf[rdsz - 1] == '\n') --rdsz;
-            buf[rdsz] = 0;
-            __android_log_write(ANDROID_LOG_DEBUG, "StdoutLogger", buf);
+            char* line = nullptr;
+            size_t n = 0;
+            ssize_t nread = getline(&line, &n, stream);
+            if (nread == -1)
+            {
+                break;
+            }
+
+            line[nread - 1] = '\0';
+            __android_log_write(prio, "StdoutLogger", line);
+            free(line);
```

Comment:
a free without a malloc.... 70's API :)

Reply:
I wish I could pass in my own buffer to avoid constant allocations, but it will call `realloc` if the size doesn't fit.